bluepysnap/circuit_validation.py

```diff
@@ -740,11 +740,117 @@ def validate(config_file, bbp_check=False):
     Returns:
         list: List of errors, empty if no errors
     """
-    config = Config(config_file).resolve()
+    config = Config(config_file)
+    if bbp_check:
+        validate_schemas(config)
+
+    config = config.resolve()
+
     errors = _check_required_datasets(config)
     if not errors:
         errors = _check_populations(config)
     if not bbp_check:
         errors = [e for e in errors if not isinstance(e, BbpError)]
     _print_errors(errors)
     return set(errors)
+
+
+def get_h5_structure_as_dict(h5):
+    """Recursively translates h5 file into a dictionary.
+
+    For groups, the subgroups/datasets are translated as a subdictionary.
+    Datatype of datasets is resolved and returned as {'datatype': <dtype>}.
+    Attributes of either groups or datasets are returned as {'attributes': {<key>: <value>}}.
+
+    Args:
+        h5 (h5.File instance): h5 file to translate
+
+    Returns:
+        dict: dictionary of the file structure
+    """
+    properties = {}
+
+    def get_dataset_dtype(item):
+        if item.dtype.hasobject:
+            return h5py.check_string_dtype(item.dtype).encoding
+
+        return item.dtype.name
+
+    for key, value in h5.items():
+        if isinstance(value, h5py.Group):
+            properties[key] = get_h5_structure_as_dict(value)
+        elif isinstance(value, h5py.Dataset):
+            properties[key] = {'datatype': get_dataset_dtype(value)}
+        else:
+            properties[key] = {}
+
+        attrs = {k: v for k, v in value.attrs.items()}
+        if attrs:
+            properties[key]['attributes'] = attrs
+
+    # Maybe check for the existence of @library and do the translation to actual values here?
+    # Would make the schemas a bit simpler.
+
+    return properties
+
+
+def validate_schemas(config):
+    def _load_schema(filename):
+        import yaml
+        import pkg_resources
+        resource_path = Path(pkg_resources.resource_filename(__name__, 'schemas'))
+
+        with open(resource_path / filename) as fd:
+            return yaml.safe_load(fd)
+
+    def _load_json(path):
+        import json
+        with open(path) as fd:
+            return json.load(fd)
+
+    def _validate(schema, dict_):
+        import jsonschema
+        validator = jsonschema.validators.Draft202012Validator(schema)
+        # currently just printing the errors but the idea would be to have something like
+        # [heavily pseudo code]
+        # errors[field] = e.type
+        # if e.type != 'warning':
+        #   failed_fields.append(e.absolute_path)
+        # ... later ...
+        # validate_data(data_file, file_structure - failed_fields)
+        for e in validator.iter_errors(dict_):
```

Comment:
yeah, agree about adding the errors to a list that returned to the caller is the best way